datumaro/plugins/synthetic_images_plugin/image_generator.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from multiprocessing import Pool
+from shutil import rmtree
+from typing import List, Optional
+import os
+
+import cv2 as cv
+import numpy as np
+
+from datumaro.components.cli_plugin import CliPlugin
+from datumaro.components.errors import SourceExistsError
+
+from .utils import IFSFunction, augment, colorize, download_colorization_model
+
+
+class ImageGenerator(CliPlugin):
+    """
+    ImageGenerator generates a 3-channel synthetic dataset with provided shapes.
```

Comment:
1. As far as I can see, only one shape is provided.
2. I don't think it's valid to say that a _dataset_ has channels or shapes. It's _images_ that have these characteristics.

Reply:
Fixed